pkg/features/embedded/grub-config/etc/elemental/grub.cfg

```diff
@@ -48,39 +48,58 @@ insmod gfxterm
 insmod loopback
 insmod squash4
 
-set loopdev="loop0"
+## Sets a loopback device volume for a given image
+function set_loopdevice {
+  set volume="loop"
+  if [ -e (${volume}) ]; then 
+    loopback -d ${volume}
+  fi
+  loopback ${volume} ${1}
+}
+
+## Defines the volume and image to boot from for active or passive boots
+function set_volume {
+  if [ "${snapshotter}" == "btrfs" ]; then
+    set btrfs_relative_path="y"
+    set volume="${root}"
+    if [ -n "${1}" ]; then
+      set img=${1}
+      btrfs-mount-subvol ($root) / @/.snapshots/${1}/snapshot
+    fi
+  elif [ -z "${1}" ]; then
+    set_loopdevice /.snapshots/active
+  else
+    set img=${1}
+    set_loopdevice /.snapshots/${1}/snapshot.img
+  fi
+}
 
 menuentry "${display_name}" --id active {
-  set img=/.snapshots/active
   set mode=active
-  search --no-floppy --label --set=root $state_label
-  loopback $loopdev $img
-  source ($loopdev)/etc/cos/bootargs.cfg
-  linux ($loopdev)$kernel $kernelcmd ${extra_cmdline} ${extra_active_cmdline}
-  initrd ($loopdev)$initramfs
-  loopback -d $loopdev
+  search --no-floppy --label --set=root ${state_label}
+  set_volume
+  source (${volume})/etc/cos/bootargs.cfg
+  linux (${volume})${kernel} ${kernelcmd} ${extra_cmdline} ${extra_active_cmdline}
+  initrd (${volume})${initramfs}
 }
 
 for passive_snap in ${passive_snaps}; do
```

Comment:
passive_snaps includes a space separated list of passive snapshots IDs. Then the path of the image is computed in set_volume function above.